### PR TITLE
feat(enhancement): Add function for proper credit string formatting

### PR DESCRIPTION
**Pull request by tibetiroka**
**Feature:** This PR implements the feature requested by @warp-core in #the-hub.

## Feature Details
Adds a new function to Format that appends the credit/credits suffix to the formatted credit amount. This fixes a million minor formatting bugs and reduces the amount of duplicate code.

## Testing Done
Checked the shipyard, jobs, salaries. Formatting works as expected.

### CODE REVIEW

File: source/text/Format.cpp [hunk 1]
```diff
@@ -93,6 +93,19 @@ string Format::Credits(int64_t value)
...
+// Convert the given number into abbreviated format as described in Format::Credits,
+// then attach the ' credit' or ' credits' suffix to it.
+string Format::CreditString(int64_t value)
+{
+	string creditValue = Credits(value);
+	if(value == 1)
+		creditValue += " credit";
+	else
+		creditValue += " credits";
+	return creditValue;
```
Comment (warp-core) [inline source/text/Format.cpp:105]:
```suggestion
	if(value == 1)
		return "1 credit";
	else
		return Credits(value) + " credits";
```

**Review by warp-core — COMMENTED** [review]
We should probably add unit tests for this as well.

**Comment by tibetiroka** [comment]
Added some unit tests for `Format::Credits` and `Format::CreditString`.

Also, am I the only one who thinks `1,000.000M` looks worse than just `1B`? (Have a look at the tests for these edge cases, I think they are kind of ugly.)

File: source/text/Format.cpp [hunk 2]
```diff
@@ -93,6 +93,17 @@ string Format::Credits(int64_t value)
...
+// then attach the ' credit' or ' credits' suffix to it.
+string Format::CreditString(int64_t value)
+{
+	if(value == 1)
+		return "1 credit";
+	else
+		return Credits(value) + " credits";
+}
+
+
```
Comment (Hurleveur) [inline source/text/Format.cpp:106]:
```suggestion


```
forgot a space

**Review by Hurleveur — COMMENTED** [review]
this simplifies the code a lot more than what I thought, nice work

**Review by Hurleveur — APPROVED** [review]
since that new test runs worse that could have happened is we forgot one, which wouldnt be the end of the world
I suppose you used a search function so you wouldn't miss any, too

edit: I meant I suppose you used the search function of an IDE instead of looking at them one by one, meaning you shouldn't have missed any.

**Comment by tibetiroka** [comment]
Sorry, what do you mean?

**Review by warp-core — APPROVED** [review]
